libs/utils/WordPressUtils/src/test/java/org/wordpress/android/util/LogFileHelpersTest.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import android.os.Build
+import androidx.test.core.app.ApplicationProvider
+import java.io.File
+import java.util.UUID
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.wordpress.android.util.helpers.logfile.LogFileProvider
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.O_MR1])
+class LogFileHelpersTest {
+    lateinit var testProvider: LogFileProvider
+
+    @Before
+    fun setup() {
+        val context: Context = ApplicationProvider.getApplicationContext()
+        testProvider = LogFileProvider.fromContext(context)
+    }
+
+    @After
+    fun tearDown() {
+        // Delete the test directory after each test
+        testProvider.getLogFileDirectory().deleteRecursively()
+    }
+
+    @Test
+    fun testThatLogFileDirectoryIsCreatedIfNotExists() {
+        val directory = testProvider.getLogFileDirectory()
+        assert(directory.exists())
+    }
+
+    @Test
+    fun testThatLogFilesListsAllFiles() {
+        val directory = testProvider.getLogFileDirectory()
+        File(directory, UUID.randomUUID().toString()).createNewFile()
+        Assert.assertEquals(testProvider.getLogFiles().count(), 1)
+    }
+
+    @Test
+    fun testThatLogFilesSortsFilesWithMostRecentFirst() {
+        val directory = testProvider.getLogFileDirectory()
+
+        var oldFile = File(directory, UUID.randomUUID().toString())
+        oldFile.createNewFile()
+        oldFile.setLastModified(1_000L)
+        assert(oldFile.lastModified() == 1_000L)
+
+        var newFile = File(directory, UUID.randomUUID().toString())
+        newFile.createNewFile()
+        newFile.setLastModified(1_000_000L)
+        assert(newFile.lastModified() == 1_000_000L)
+
+        val files = testProvider.getLogFiles()
+        assert(files.first().lastModified() < files.last().lastModified())
```

Comment:
I am just geeking it out here since you mentioned you enjoyed nice functional programming bits, so here is a pretty cool way to write this:

```suggestion
            assert(testProvider.getLogFiles().asSequence()
                .zipWithNext { a, b -> a.lastModified() < b.lastModified()}
                .all { it })
```

This will check if all files are in order, so by combining it with the previous suggestion, you can actually test more modified dates.

Reply:
This _is_ cool – to test with it properly, it'd probably make sense to rework the file generation stuff to be more like:

```kotlin
        generateSequence { Random.nextLong(0, 1000) }
                .take(Random.nextInt(2,10)) // no more than 10 files
                .forEach {
                    File(directory, UUID.randomUUID().toString()).also { file ->
                        // don't test the file system's date resolution – use timestamps in increments of 1000
                        val date = it * 1000 
                        file.createNewFile()
                        file.setLastModified(date)
                        assert(file.lastModified() == date)
                    }
                }
```

But because this already uses `UUID.randomUUID()` as the filename, adding a quick `shuffled` to the timestamp array ensures that we'll be randomizing the order of all the relevant elements (filename, creation date, modification date) to test the implementation, so it's probably nicer to future readers to just keep the test code simple 😄 

Updated in 1645984007f4980309c8bf45e9b6faba80c54d81

Reply:
Well, we don't have to change the file generation, but I do agree the current solution is very simple, so 👍